tests/unit/test-cmd/test.lint.js

```diff
@@ -127,20 +126,22 @@ describe('lint', () => {
     });
   });
 
-  it('passes a file filter to the linter', () => {
-    const fileFilter = fake(new FileFilter());
-    const {lint, createLinter} = setUp({fileFilter});
-    return lint()
-      .then(() => {
-        assert.equal(createLinter.called, true);
-        const config = createLinter.firstCall.args[0].config;
-        assert.isFunction(config.shouldScanFile);
-
-        // Simulate how the linter will use this callback.
-        config.shouldScanFile('manifest.json');
-        assert.equal(fileFilter.wantFile.called, true);
-        assert.equal(fileFilter.wantFile.firstCall.args[0], 'manifest.json');
-      });
+  it('ensure linter build fileFilter and shouldScanFile correctly', () => {
+    const fileFilter = {wantFile: sinon.spy(() => true)};
+    const createFileFilter = sinon.spy(() => fileFilter);
+    const {lint, createLinter} = setUp({createFileFilter});
+    const params = {
+      sourceDir: '.',
+      artifactsDir: 'artifacts',
+      ignoreFiles: ['file1', '**/file2'],
+    };
+    return lint(params).then(() => {
+      assert(createFileFilter.calledWithMatch(params));
+      assert(createLinter.called);
+      const {shouldScanFile} = createLinter.firstCall.args[0].config;
+      shouldScanFile('path/to/file');
+      assert(fileFilter.wantFile.calledWith('path/to/file'));
```

Comment:
this should change to the `assert.equal(fileFilter.wantFile.firstCall.args[0], ...)` pattern too